package/MDAnalysis/lib/nsgrid.pyx

```diff
@@ -840,82 +457,58 @@ cdef class FastNS(object):
         if any of the query coordinates lies outside the `box` supplied to
         :class:`~MDAnalysis.lib.nsgrid.FastNS`.
         """
-
-        cdef ns_int i, j, size_search
-        cdef ns_int d, m
-        cdef ns_int current_beadid, bid
-        cdef ns_int cellindex, cellindex_probe
-        cdef ns_int xi, yi, zi
-
+        from MDAnalysis.lib import distances
+
+        cdef int i, j, size_search
+        cdef int cx, cy, cz
+        cdef int cellid
+        cdef int xi, yi, zi
+        cdef int cellcoord[3], searchcoord[3]
+        
         cdef NSResults results
-
-        cdef dreal d2
-        cdef rvec probe
-
-        cdef real[:, ::1] searchcoords
-        cdef real[:, ::1] searchcoords_bbox
-        cdef _NSGrid searchgrid
-        cdef bint check
-
-        cdef dreal cutoff2 = self.cutoff * self.cutoff
-        cdef ns_int npairs = 0
+        cdef double d2, cutoff2
+        cdef float[:, ::1] searchcoords_bbox
+        
+        cutoff2 = self.cutoff * self.cutoff
 
         if (search_coords.ndim != 2 or search_coords.shape[1] != 3):
             raise ValueError("search_coords must have a shape of (n, 3), got "
                              "{}.".format(search_coords.shape))
 
-        # Generate another grid to search
-        searchcoords = search_coords.astype(np.float32, order='C', copy=False)
-        searchcoords_bbox = self.box.fast_put_atoms_in_bbox(searchcoords)
-        searchgrid = _NSGrid(searchcoords_bbox.shape[0], self.grid.used_cutoff, self.box, self.max_gridsize, force=True)
-
-        size_search = searchcoords.shape[0]
+        searchcoords_bbox =  distances.apply_PBC(search_coords, self.grid.dimensions)
```

Comment:
This is problematic for non-PBC calculations if coordinates lie outside the pseudobox.